framework/sandstone.cpp

```diff
@@ -739,6 +740,16 @@ bool test_is_retry() noexcept
     return sApp->current_iteration_count < 0;
 }
 
+bool test_inject_idle(int N) noexcept
+{
+    if (sApp->shmem->current_test_sleep_duration > 0) {
+        usleep(sApp->shmem->current_test_sleep_duration / N);
```

Comment:
While working on the Windows changes, I found a very old commit of mine that adds a `clock_sleep_for(Duration d)`. I think it's worthwhile resurrecting it.

I can do that after your change.